Buttplug.Client/ButtplugWSClient.cs

```diff
@@ -110,19 +107,23 @@ public ButtplugWSClient(string aClientName)
                 throw new AccessViolationException("Already connected!");
             }
 
-            _ws = new ClientWebSocket();
+            _ws = new WebSocket(aURL.ToString());
             _waitingMsgs.Clear();
             _devices.Clear();
             _counter = 1;
-            await _ws.ConnectAsync(aURL, CancellationToken.None);
+            _ws.Open();
+
+            while (_ws.State == WebSocketState.Connecting)
```

Comment:
According to the examples in WebSocket4Net's tests (https://github.com/kerryjiang/WebSocket4Net/blob/master/Test/Test/WebSocketClientTest.cs#L199), there's Open events. Why spin-wait when these events could be hooked up and possibly awaited, giving us basically the same interface we had before?